### ISSUE TITLE
tensorflow + numpy compatibility?

### ISSUE THREAD

**Issue by jiapei100** [tensorflow/tensorflow#31249]
```console
➜  ~ pip show tensorflow
pipName: tensorflow
Version: 1.14.0
➜  ~ pip show numpy
Name: numpy
Version: 1.17.0
```

```console
>>> import tensorflow as tf
~/.local/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
~/.local/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
~/.local/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
~/.local/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
~/.local/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:545: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
~/.local/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:550: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])

```

**Comment by ymodak** [comment]
Do you have multiple versions of numpy installed in your system?
```pip show numpy```
Can you uninstall all numpy versions and later install numpy==1.16.4?
```pip uninstall numpy``` (till you uninstall all versions)
``` pip install numpy==1.16.4```

**Comment by jiapei100** [comment]
@ymodak 

I manually upgraded **numpy**:

```console
pip install -U numpy --user
```
to **~/.local/lib/python3.7/site-packages**

And, yes, I happened to notice I've got **python3-numpy** installed as well, from Ubuntu repository. 

Did tensorflow find **python3-nump** from repository and had it installed automatically? I specified PYTHONPATH to **~/.lcoal/lib/python3/site-packages** already.

Cheers

**Comment by DSMLAKGER** [comment]
@ymodak :
Same issue for me after fresh python and tensorflow installation (tensorflow 1.14.0 and numpy 1.17.0) from today morning.
had only one numpy installation. but after uninstallation of 1.17.0 and installation of 1.16.4 the Future Warnings are gone. Thank you!

**Comment by gadagashwini** [comment]
@jiapei100  Can you please let us know if you are happy to close if no issue persists. Thanks!

**Comment by vrsheen** [comment]
For me the warning got resolved after doing the below 2:
1) >pip install scikit-learn
2) >pip install tensorflow-datasets

**Comment by fatemehtorki** [comment]
@ymodak 
Same issue happened to me with (tensorflow  1.9.0 and numpy 1.17.0) 
 after uninstall of 1.17.0 and installation of 1.16.4 the Future Warnings are gone. Thank you so much!

Can you help me for solve another issue?
after  i load model ,when i predicted this error is occur (Tensor Tensor("dense_2/Softmax:0", shape=(?, 2), dtype=float32) is not an element of this graph) .
What do you think I should do...

**Comment by IanQS** [comment]
@fatemehtorki

You should open a new issue for that error since it is unrelated to this error

**Comment by pedro-w** [comment]
Please can you advise, it it the intention to update Tensorflow 1.x so that these warnings won't occur in numpy 1.17 and later? Or will we always have to stick at 1.16.4?

**Comment by mihaimaruseac** [comment]
See #30559 Next release will have the issue solved

**Comment by sudonto** [comment]
@fatemehtorki 
You may ask in stackoverflow for that

**Comment by Mubasher0331** [comment]
hello i have the same issue as jiapi i have the same versions of numpy and tensor flow please tell its solution

**Comment by Frozoto1** [comment]
@ymodak It's helpful, thank you!

**Comment by thehayat** [comment]
I'm a bit new @ymodak @jiapei100.
I have 
`numpy 1.16.4
tensorflow 1.14.0
tensorflow-gpu 1.14.0
keras 2.2.4
keras -Applications 1.0.8
Keras-Preprocessing 1.1.0`
I am trying CNN and my jupyter is crashing on executing.
`bottleneck_features_train = vgg16.predict_generator(generator, predict_size_train)`

Please help I think its a compatibility issue. 

Full code [link](https://github.com/imamun93/animal-image-classifications/blob/master/final_notebook.ipynb).


![image](https://user-images.githubusercontent.com/29543170/63399741-1b0dcc80-c3ef-11e9-909b-898e456d4e13.png)

**Comment by ganesh-ms** [comment]
> I'm a bit new @ymodak @jiapei100.
> I have
> `numpy 1.16.4 tensorflow 1.14.0 tensorflow-gpu 1.14.0 keras 2.2.4 keras -Applications 1.0.8 Keras-Preprocessing 1.1.0`
> I am trying CNN and my jupyter is crashing on executing.
> `bottleneck_features_train = vgg16.predict_generator(generator, predict_size_train)`
> 
> Please help I think its a compatibility issue.
> 
> Full code [link](https://github.com/imamun93/animal-image-classifications/blob/master/final_notebook.ipynb).
> 
> ![image](https://user-images.githubusercontent.com/29543170/63399741-1b0dcc80-c3ef-11e9-909b-898e456d4e13.png)

Your Tensorflow is not properly installed. Try installing it again.

**Comment by SamiEzz** [comment]
the same problem here the answer https://stackoverflow.com/a/58478238/12248686

**Comment by SamiEzz** [comment]
I had the same problem in my linux laptop with tensorflow in python3 v3.6
Actually, you just need to change some lines in 2 files :
- 1

`~/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py

now change this code : (line 516)
```

_np_qint8 = np.dtype([("qint8", np.int8, 1)])
_np_quint8 = np.dtype([("quint8", np.uint8, 1)])
_np_qint16 = np.dtype([("qint16", np.int16, 1)])
_np_quint16 = np.dtype([("quint16", np.uint16, 1)])
_np_qint32 = np.dtype([("qint32", np.int32, 1)])


# _np_bfloat16 is defined by a module import.

# Custom struct dtype for directly-fed ResourceHandles of supported type(s).
np_resource = np.dtype([("resource", np.ubyte, 1)])
```
by this code :
```

_np_qint8 = np.dtype([("qint8", np.int8, (1,))])
_np_quint8 = np.dtype([("quint8", np.uint8, (1,))])
_np_qint16 = np.dtype([("qint16", np.int16, (1,))])
_np_quint16 = np.dtype([("quint16", np.uint16, (1,))])
_np_qint32 = np.dtype([("qint32", np.int32, (1,))])

# _np_bfloat16 is defined by a module import.

# Custom struct dtype for directly-fed ResourceHandles of supported type(s).
np_resource = np.dtype([("resource", np.ubyte, (1,))])
```
you have to do the same on this file :

`~/.local/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py`


wubbalubadubdub

**Comment by mattc-eostar** [comment]
@SamiEzz 
Is this dependent on a particular version of numpy? I am currently using 1.16.4, but 1.17.0 should be fine with TF2.0.0 right?

**Comment by SamiEzz** [comment]
> @SamiEzz
> Is this dependent on a particular version of numpy? I am currently using 1.16.4, but 1.17.0 should be fine with TF2.0.0 right?

I had this issue with the latest version of numpy ('1.17.3')

**Comment by gadagashwini** [comment]
@mattc-eostar and @SamiEzz, Please use numpy 1.16.5 version. Thanks!

**Comment by michalisfrangos** [comment]
Thought to mention a related issue #30120: 

The use of the numpy (1.17.3) together and latest tf-nightly,  causes the error of discussion in #30120  

- **SOLUTION** :  install previous version of numpy    1.16.4

- tensorlfow version : 2.1.0-dev20191023 
- numpy:  1.17.3 (install the previous version 1.16.4)
-  python: 3.7

**Comment by Adnan-Toky** [comment]
I was facing the same issue. Downgrading numpy to version 1.16.4 solved the problem. Thanks @ymodak.

**Comment by DanyEle** [comment]
Got the same issue with numpy==1.18.0. Downgrading to numpy 1.16.4 solved the issue for me:

`pip3 install numpy==1.16.4 `

**Comment by eric-wieser** [comment]
For reference, the numpy change which causes this warning in tensorflow is https://github.com/numpy/numpy/pull/13326

The warning is harmless to end users, and there is no need to downgrade. Aside from emitting a warning, the behavior has not changed.

**Comment by ArthurYdalgo** [comment]
> Got the same issue with numpy==1.18.0. Downgrading to numpy 1.16.4 solved the issue for me:
> 
> `pip3 uninstall numpy`
> `pip3 install numpy==1.16.4 `

this solved for me :)

**Comment by mihaimaruseac** [comment]
Locking as issue has been already solved. If there are new instances of issue and no solution here helps, please open a new issue, filling in issue template and documenting why these steps don't work.
